charts/core/src/Echart/useEchart.ts

```diff
@@ -0,0 +1,386 @@
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import debounce from 'lodash.debounce';
+
+import { ChartOptions } from '../Chart';
+import { chartSeriesColors } from '../Chart/chartSeriesColors';
+import * as updateUtils from '../Chart/hooks/updateUtils';
+
+import {
+  EChartHookProps,
+  EChartsInstance,
+  ZoomSelectionEvent,
+} from './Echart.types';
+
+// Singleton promise for initialization to prevent duplication
+let initPromise: Promise<void> | null = null;
+let echartsCore: any;
+let echartsCharts: any;
+let echartsRenders: any;
+let echartsComponents: any;
+
+async function initializeEcharts() {
+  // If already initialized, return immediately
+  if (echartsCore) {
+    return;
+  }
+
+  // If initialization is in progress, wait for it
+  if (initPromise) {
+    return initPromise;
+  }
+
+  // Start initialization and store the promise
+  initPromise = (async () => {
+    try {
+      const [
+        echartsCoreResolved,
+        echartsChartsResolved,
+        echartsRendersResolved,
+        echartsComponentsResolved,
+      ] = await Promise.all([
+        import('echarts/core'),
+        import('echarts/charts'),
+        import('echarts/renderers'),
+        import('echarts/components'),
+      ]);
+
+      echartsCore = echartsCoreResolved;
+      echartsCharts = echartsChartsResolved;
+      echartsRenders = echartsRendersResolved;
+      echartsComponents = echartsComponentsResolved;
+
+      echartsCore.use([
+        echartsCharts.LineChart,
+        echartsRenders.CanvasRenderer,
+        echartsComponents.TitleComponent,
+        echartsComponents.TooltipComponent,
+        echartsComponents.GridComponent,
+        echartsComponents.LegendComponent,
+        echartsComponents.ToolboxComponent,
+        echartsComponents.DataZoomComponent,
+        echartsComponents.DataZoomInsideComponent,
+      ]);
+    } catch (error) {
+      // Ensure we clear the promise if initialization fails
+      initPromise = null;
+      throw error;
+    }
+  })();
+
+  return initPromise;
```

Comment:
Should this return the modules themselves?

Reply:
Updated to return the core module since it's the only one used outside of this method.